scanomatic/models/validators/validate.py

```diff
@@ -0,0 +1,92 @@
+from collections.abc import Iterator
```

Comment:
This is a reimplementation of previous validation logic and should be considered new code